app/src/main/java/com/keylesspalace/tusky/components/notifications/NotificationFetcher.kt

```diff
@@ -65,23 +66,24 @@ class NotificationFetcher @Inject constructor(
                         }
                     }
 
+                    val notificationsGrouped = groupByGroupId(account, notifications)
+
                     // Make and send the new notifications
                     // TODO: Use the batch notification API available in NotificationManagerCompat
                     // 1.11 and up (https://developer.android.com/jetpack/androidx/releases/core#1.11.0-alpha01)
                     // when it is released.
-                    notifications.forEachIndexed { index, notification ->
-                        val androidNotification = NotificationHelper.make(
-                            context,
-                            notificationManager,
-                            notification,
-                            account,
-                            index == 0
-                        )
-                        notificationManager.notify(notification.id, account.id.toInt(), androidNotification)
-                        // Android will rate limit / drop notifications if they're posted too
-                        // quickly. There is no indication to the user that this happened.
-                        // See https://github.com/tuskyapp/Tusky/pull/3626#discussion_r1192963664
-                        delay(1000.milliseconds)
+
+                    notificationsGrouped.forEach { notificationList ->
+                        notificationList.forEach { notification ->
+                            val androidNotification = NotificationHelper.make(
+                                context,
+                                notificationManager,
+                                notification,
+                                account,
+                                notificationList.size == 1
+                            )
+                            notificationManager.notify(notification.id, account.id.toInt(), androidNotification)
```

Comment:
Why remove the delay? This will result in notifications being dropped.

Reply:
See my findings and the documentation link in https://github.com/tuskyapp/Tusky/issues/3756#issuecomment-1605342221

The only thing that drops (grouped) notifications for me is the maximum total number.

Reply:
Right, that's what confuses me. Because in that comment you note that you saw this message being logged if the delay is removed.

> Package has already posted or enqueued 25 notifications. Not showing more. package=com.keylesspalace.tusky"

So, remove the delay and Android will refuse to show some of the notifications.

Reply:
My findings were from my device (and Android 10). With the emulator Logcat shows something like "Silencing noisy Tusky" as an error after 15 or so (group) notifications without delay.

If they are actually "dropped" I need to check again, also on my device.

However dropping we will have in any case for lots of notifications: a) in our code (50) and b) on actual devices (25 on mine).

So maybe we can (must) leave the delay in here.

But we should (must) switch to batched notifications as soon as possible, I think.

Reply:
Added delay again. It might help on some devices.